iroha/src/queue.rs

```diff
@@ -66,29 +66,43 @@ impl Queue {
                 .pending_tx_by_hash
                 .get(&transaction_hash)
                 .expect("Failed to get tx by hash.");
-            if !transaction.is_expired(self.transaction_time_to_live) {
-                if is_leader {
-                    if transaction
-                        .check_signature_condition(world_state_view)
-                        .map_or(false, |b| b)
-                    {
+            if !transaction.is_expired(self.transaction_time_to_live)
+                && !transaction.is_in_blockchain(world_state_view)
+            {
+                if let Ok(signature_condition_passed) =
+                    transaction.check_signature_condition(world_state_view)
+                {
+                    if is_leader {
+                        if signature_condition_passed {
+                            output_transactions.push(
+                                self.pending_tx_by_hash
+                                    .get(&transaction_hash)
+                                    .expect("Failed to get tx by hash.")
```

Comment:
Is it okay to panic here? May be `.expect` should contain message why `.get` never fails?

Reply:
Yes, get should never fail here by design. But if it fails, to my mind it is rather important to get a message at what it failed to correct this bug.

Reply:
Ok. Then a comment with contract why it never fails would be handy right?